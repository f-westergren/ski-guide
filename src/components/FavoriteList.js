import React, { useState, useEffect } from 'react';
import { Spinner, Container } from 'reactstrap';
import FavoriteCard from './FavoriteCard';
import SkiGuideApi from '../SkiGuideApi';

const FavoriteList = () => {
  const [isLoading, setIsLoading] = useState(true);
  const [isError, setIsError] = useState(false);
  const [favorites, setFavorites] = useState([])
  
  useEffect(() => {
    
      const fetchFavorites = async () => {
        try {
          const res = await SkiGuideApi.getFavorites();
          setFavorites(res.favorites)
          setIsLoading(false);
        } catch (err) {
          setIsError(true)
          setIsLoading(false);
        }
      }
      fetchFavorites();
  }, [])
  
  if (isLoading) {
    return (
      <div className="text-center mt-5">
        <Spinner animation="border" role="status" />
        <span className="sr-only">Loading...</span>
      </div>
    )
  }

  if (isError) {
    return (
      <h4 className="text-white text-center mt-5">Can't get favorites.</h4>
    )
  }

  return (
    <Container className='profile mt-5 p-4 main-container'>
      <h2 className="text-center">Favorites</h2>
      {!favorites.length && <p className="text-center">You have no favorites yet :(</p>}
      {favorites.map(fav => (
        <FavoriteCard 
          first_name={fav.first_name}
          image_url={fav.image_url}
          location={fav.location}
          guideId={fav.guide_id}
          key={fav.id}
          id={fav.id}
        />
      ))}
    </Container>
  )
}

export default FavoriteList