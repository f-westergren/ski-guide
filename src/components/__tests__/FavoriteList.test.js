import React from "react";
import { render } from "@testing-library/react";
import FavoriteList from "../FavoriteList";


it("renders without crashing", function() {
  render(<FavoriteList />);
});

it("matches snapshot", function() {
  const { asFragment } = render(<FavoriteList />);
  expect(asFragment()).toMatchSnapshot();
});
