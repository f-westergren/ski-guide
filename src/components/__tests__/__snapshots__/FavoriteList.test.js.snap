// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="text-center mt-5"
  >
    <div
      animation="border"
      class="spinner-border"
      role="status"
    >
      <span
        class="sr-only"
      >
        Loading...
      </span>
    </div>
    <span
      class="sr-only"
    >
      Loading...
    </span>
  </div>
</DocumentFragment>
`;
